File: dart_lang/lib/a_solid/d_interface_segregation/a_pre_refactor/subscriber.dart
import 'package:dart_lang/a_solid/d_interface_segregation/a_pre_refactor/customer.dart';
import 'package:dart_lang/a_solid/d_interface_segregation/a_pre_refactor/user_manager.dart';

class Subscriber implements UserManager {
  @override
  void changePassword(Customer customer, String newPassword) {}

  @override
  void subscribeToMSNotifications() {}

  @override
  void subscribeToNewProductsAvailability() {
    print("subscribing ...");
  }

  @override
  void updateUserProfile(Customer customer) {
    print("updating");
  }
}

File: PHP/Solid Principles/interface_segregation/problem/Subscriber.php
<?php

class Subscriber implements UserManager {
    public function changePassword($customer, $newPassword): void {}

    public function subscribeToMSNotifications(Customer $customer): void {}

    public function subscribeToNewProductsAvailability(Customer $customer): void {
        echo "subscribing ...";
    }

    public function updateUserProfile($customer): void {
        echo "updating";
    }
}